ReSwift/CoreTypes/Store.swift

```diff
@@ -28,7 +28,7 @@ open class Store<State: StateType>: StoreType {
             subscriptions.forEach {
                 // if a selector is available, subselect the relevant state
                 // otherwise pass the entire state to the subscriber
-                $0.subscriber?._newState(state: $0.selector?(state) ?? state)
+                $0.subscriber?._newState(state: $0.selector?(state) ?? state!)
```

Comment:
actually thats not true either... Looking into it further. I've got a test case with a nil reducer that passes with the current code `$0.subscriber?._newState(state: $0.selector?(state) ?? state)`, but crashes with both `!` and `as any`...


```
    /**
     it allows the app reducer to return nil
     */
    func testAllowsReducerReturningNil() {
        let reducer: Reducer<TestAppState> = { action, state in
            return nil
        }
        store = Store(reducer: reducer, state: TestAppState())
        let action = SetValueAction(3)
        let subscriber = TestSubscriber()

        store.subscribe(subscriber)
        store.dispatch(action)

        XCTAssertNil(store.state)
        XCTAssertEqual(subscriber.receivedStates.count, 1)
        guard let last = subscriber.receivedStates.last else {
            XCTFail("Received state count 1, but could not fetch last")
            return
        }
        XCTAssertNil(last.testValue)
    }
```